anta/tests/snmp.py

```diff
@@ -237,3 +249,73 @@ def test(self) -> None:
             self.result.is_failure(f"Expected `{self.inputs.contact}` as the contact, but found `{contact}` instead.")
         else:
             self.result.is_success()
+
+
+class VerifySNMPErrors(AntaTest):
+    """Verifies the number of SNMP error counter(s) processed.
+
+    By default, all  error counters will be checked for any non-zero values.
+    An optional list of specific error counters can be provided for granular testing.
+
+    Expected Results
+    ----------------
+    * Success: The test will pass if the SNMP error counter(s) are zero/None.
+    * Failure: The test will fail if the SNMP error counter(s) are non-zero/not None/Not Found or is not configured.
+
+    Examples
+    --------
+    ```yaml
+    anta.tests.snmp:
+      - VerifySNMPErrors:
+          error_counters:
+            - inVersionErrs
+            - inBadCommunityNames
+    """
+
+    name = "VerifySNMPErrors"
+    description = "Verifies the number of SNMP error counter(s) processed."
+    categories: ClassVar[list[str]] = ["snmp"]
+    commands: ClassVar[list[AntaCommand | AntaTemplate]] = [AntaCommand(command="show snmp", revision=1)]
+
+    class Input(AntaTest.Input):
+        """Input model for the VerifySNMPErrors test."""
+
+        error_counters: (
+            list[
+                Literal[
+                    "inVersionErrs",
+                    "inBadCommunityNames",
+                    "inBadCommunityUses",
+                    "inParseErrs",
+                    "outTooBigErrs",
+                    "outNoSuchNameErrs",
+                    "outBadValueErrs",
+                    "outGeneralErrs",
+                ]
+            ]
+            | None
+        ) = None
+        """Optional list of SNMP error counters to be verified. If not provided, test will verifies all the counters."""
+
+    @AntaTest.anta_test
+    def test(self) -> None:
+        """Main test function for VerifySNMPPDUs."""
+        error_counters = self.inputs.error_counters
+        command_output = self.instance_commands[0].json_output
+
+        # Verify SNMP PDU counters.
+        if not (snmp_counters := get_value(command_output, "counters")):
+            self.result.is_failure("SNMP counter details not found.")
+            return
+
+        # In case SNMP error counters not provided, It will check all the error counters.
+        if not error_counters:
```

Comment:
same as other PR, iterate on actual command output not on defined list

Reply:
We need to iterate only defined error counters as in actual output we have error counters and trapPDUs also.

Reply:
Same as other PR, you can use `typing.get_args` here.